src/signTx.c

```diff
@@ -198,11 +204,29 @@ static void signTx_handleInit_ui_runStep()
 
 	UI_STEP_BEGIN(ctx->ui_step);
 	UI_STEP(HANDLE_INIT_STEP_DISPLAY_DETAILS) {
-		ui_displayNetworkParamsScreen(
-		        "New transaction",
-		        ctx->commonTxData.networkId, ctx->commonTxData.protocolMagic,
-		        this_fn
-		);
+		if (ctx->numOutputs > 0) {
```

Comment:
HANDLE_INIT_STEP_DISPLAY_DETAILS is triggered when there is some kind of mismatch between protocol magic an network id. A simple fix would be adding the same condition being checked here into the policy. A caveat though is that If there is actually a mismatch between network id and protocol magic, this UI logic would skip the displaying of network id/protocol magic if the tx has a withdrawal and no outputs (where network id is actually verifiable)

https://github.com/vacuumlabs/ledger-app-cardano-shelley/blob/5f602944dd268e6363474c21858a4136f5501f39/src/securityPolicy.c#L179

Reply:
pushed commit with fix